fsm.go

```diff
@@ -67,6 +69,14 @@ func (r *fsmS) scheduleRetry(e *f.Event, cb func(e *f.Event)) {
 	}()
 }
 
+func (r *fsmS) scheduleRetryWithExponentialDelay(e *f.Event, cb func(e *f.Event), retryNumber int) {
+	r.timer = time.NewTimer(getRetryPeriodForIteration(retryNumber))
+	go func() {
+		<-r.timer.C
+		cb(e)
+	}()
+}
```

Comment:
A function that spawns a goroutine and return immediately are hard to test and they tend to be prone to leaking goroutines. I'd suggest we use a simple `time.Sleep()` here with a subsequent call to `cb()` and leave it up to the caller side to control whether to invoke this callback in a sync or async way.

Reply:
Absolutely agree, it was copy/paste of `scheduleRetry`. I will change it in a sec.